Source/events.handling/EventHandlerDecoratedTypes.ts

```diff
@@ -7,9 +7,23 @@ import { ScopeId } from '@dolittle/sdk.events';
 import { EventHandlerDecoratedType, EventHandlerId } from './index';
 
 export class EventHandlerDecoratedTypes {
-    static readonly types: ReplaySubject<EventHandlerDecoratedType> = new ReplaySubject<EventHandlerDecoratedType>();
+    private static readonly _eventHandlers = new Map<Function, EventHandlerId>();
+    private static readonly _scopes  = new Map<Function, ScopeId>();
+    static registerEventHandler(eventHandlerId: EventHandlerId, type: Function) {
+        this._eventHandlers.set(type, eventHandlerId);
+    }
+    static registerScope(scopeId: ScopeId, type: Function) {
+        this._scopes.set(type, scopeId);
+    }
```

Comment:
Probably need to handle double registration here and keep a track of the registered event handler ids so that there are no that clash

Reply:
Looking at the DotNET.SDK:
```csharp
        readonly ConcurrentDictionary<Type, List<EventHandlerType>> _eventHandlersByEventType = new ConcurrentDictionary<Type, List<EventHandlerType>>();

        public void RegisterHandler(EventHandlerId eventHandler, IEnumerable<Type> eventTypes)
        {
            eventTypes.ForEach(_ =>
                {
                    var eventHandlerType = new EventHandlerType(eventHandler, _);
                    _eventHandlersByEventType.AddOrUpdate(_, new List<EventHandlerType> { eventHandlerType }, (_, v) =>
                        {
                            v.Add(eventHandlerType);
                            return v;
                        });
                });
        }
```

 we actually have a dictionary of `<Type, List<EventHandlerType>>` where each type (`Function` for us) has a list of eventhandlers and this dictionary can be added/updated to. I'm guessing we want to mirror this same functionality here then too.

Reply:
I also don't see any specific code in the DotNET.SDK regarding handling duplicate EventhandlerId's etc but maybe I'm just not seeing it

Reply:
Registering multiple event handlers is not allowed by the Runtime, so I guess that it's up to us whether or not we want to disallow it in the SDKs as well.

The `EventHandlerType` is similar to `EventHandlerDecoratedType` that we have in the JavaScript.SDK, but it also holds additional information for registering the event handler (don't need this in DotNET since you can get all the attributes on a Type from the Type itself, while we cannot do that in JavaScript)

Reply:
I also wouldn't think that much on what's in the DotNET SDK. Since the JavaScript SDK represents the evolution that the DotNET SDK is going to go through soon. So let's spend time thinking about that first and then we can use that knowledge over in the DotNET SDK

Reply:
> I also don't see any specific code in the DotNET.SDK regarding handling duplicate EventhandlerId's etc but maybe I'm just not seeing it

On that note again. The SDK should mirror the capabilities of the Runtime. The Runtime is responsible for handling cases which can go wrong (for instance registering multiple event handlers with the same event handler id). Sone rules might be good to enforce in the SDKs while some are not that important. I think that we should have a general discussion on that in our team, on what the responsibilities of the SDKs should be @einari @jakhog My goal, at least, is to have the SDKs be as user friendly as possible, providing meaningful help along the way